internal/transport/http_util.go

```diff
@@ -329,20 +329,24 @@ func (w *bufWriter) Write(b []byte) (n int, err error) {
 		b := w.pool.Get().(*[]byte)
 		w.buf = *b
 	}
+	var bytesWritten int
 	for len(b) > 0 {
 		nn := copy(w.buf[w.offset:], b)
 		b = b[nn:]
 		w.offset += nn
-		n += nn
 		if w.offset >= w.batchSize {
-			err = w.flushKeepBuffer()
+			bytesWritten, err = w.flushKeepBuffer()
+			n += bytesWritten
+			if err != nil {
+				return n, err
+			}
```

Comment:
The tests are failing because with these new changes, `n` is not updated when the condition `if w.offset >= w.batchSize` it not `true`.

I think we can rewrite with variable names that are more descriptive (and without named return parameters) as follows:
```golang
func (w *bufWriter) Write(b []byte) (int, error) {
	if w.err != nil {
		return 0, w.err
	}
	if w.batchSize == 0 { // Buffer has been disabled.
		n, err := w.conn.Write(b)
		return n, toIOError(err)
	}
	if w.buf == nil {
		b := w.pool.Get().(*[]byte)
		w.buf = *b
	}
	written := 0
	for len(b) > 0 {
		copied := copy(w.buf[w.offset:], b)
		b = b[copied:]
		w.offset += copied
		if w.offset < w.batchSize {
			written += copied
			continue
		}
		flushed, err := w.flushKeepBuffer()
		if err != nil {
			return written + flushed, err
		}
		written += flushed
	}
	return written, nil
}
```